examples/middleware/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/iron-io/functions/api/models"
+	"github.com/iron-io/functions/api/server"
+)
+
+func main() {
+	ctx := context.Background()
+
+	funcServer := server.NewEnv(ctx)
+
+	funcServer.AddMiddlewareFunc(func(w http.ResponseWriter, r *http.Request, app *models.App) error {
+		fmt.Println("CustomMiddlewareFunc called at:", time.Now())
+		// TODO: probably need a way to let the chain go forward here and return back to the middleware, for things like timing, etc.
+		return nil
+	})
+	funcServer.AddMiddleware(&CustomMiddleware{})
+
+	funcServer.Start(ctx)
+}
+
+type CustomMiddleware struct {
+}
+
+func (h *CustomMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, app *models.App) error {
```

Comment:
Agree with you on that, I'll have to think of an example. Closing this to create a PR against master.